src/phpDocumentor/UriFactory.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor;
+
+use InvalidArgumentException;
+use League\Uri\Contracts\UriInterface;
+use League\Uri\Uri as LeagueUri;
+use Throwable;
+use const DIRECTORY_SEPARATOR;
+use function preg_match;
+use function sprintf;
+use function str_replace;
+use function strlen;
+use function strpos;
+use function substr;
+
+final class UriFactory
```

Comment:
What you could do, is instead of making this a factory, is to make this a Proxy / Decorator around `League/Uri` and have the createUri method as a static factory method. That way:

1. you don't have to inject a Factory service into the DSN
2. we have an anti-corruption layer between phpDocumentor and League/Uri; possibly making it easier to replace league/uri in the future if necessary.

If you agree, I would recommend doing that change after merging this PR